dyno-core/src/main/java/com/netflix/dyno/connectionpool/Host.java

```diff
@@ -152,6 +174,12 @@ public boolean equals(Object obj) {
             return false;
 
         Host other = (Host) obj;
+        /* we need a way to pass the information about the hashtag from the token map supplier
+         * to the host object. 
+         */
+        if (other.hashtag != null) {
+            setHashtag(other.hashtag);
+        }
         boolean equals = true;
```

Comment:
Also shouldn't Equals and compareTo be logically be the same?

Reply:
`equals()` uses both the hostname and the rack. `compareTo()` either the hostname or the rack.

Reply:
Fixed it